selfdrive/controls/lib/desire_helper.py

```diff
@@ -121,8 +109,12 @@ def update(self, carstate, lateral_active, lane_change_prob):
 
     self.prev_one_blinker = one_blinker
 
-    if self.lane_turn_direction != custom.TurnDirection.none:
-      self.desire = TURN_DESIRES[self.lane_turn_direction]
+    # Lane turn controller update
+    turn_desire = self.lane_turn_controller.update(carstate.leftBlindspot, carstate.rightBlindspot, carstate.leftBlinker, carstate.rightBlinker,
+                                                   carstate.vEgo, carstate.steeringPressed, carstate.steeringTorque)
+    self.lane_turn_direction = self.lane_turn_controller.get_lane_turn_direction()
+    if turn_desire != log.Desire.none:
+      self.desire = turn_desire
     else:
       self.desire = DESIRES[self.lane_change_direction][self.lane_change_state]
```

Comment:
Let’s keep the stock self.desire assignment as-is and place our custom logic right after it rather than wrapping it in an else block.
```suggestion
    self.desire = DESIRES[self.lane_change_direction][self.lane_change_state]

    # Lane turn controller update
    turn_desire = self.lane_turn_controller.update(carstate.leftBlindspot, carstate.rightBlindspot, carstate.leftBlinker, carstate.rightBlinker,
                                                   carstate.vEgo, carstate.steeringPressed, carstate.steeringTorque)
    self.lane_turn_direction = self.lane_turn_controller.get_lane_turn_direction()
    if turn_desire != log.Desire.none:
      self.desire = turn_desire
```